src/neuroconv/nwbconverter.py

```diff
@@ -60,15 +60,44 @@ def validate_source(cls, source_data: Dict[str, dict], verbose: bool = True):
         """Validate source_data against Converter source_schema."""
         cls._validate_source_data(source_data=source_data, verbose=verbose)
 
-    def __init__(self, source_data: Dict[str, dict], verbose: bool = True):
+    def __init__(
+        self, source_data: Dict[str, dict] = None, data_interfaces: Union[List, Dict] = None, verbose: bool = True
+    ):
         """Validate source_data against source_schema and initialize all data interfaces."""
         self.verbose = verbose
-        self._validate_source_data(source_data=source_data, verbose=self.verbose)
-        self.data_interface_objects = {
-            name: data_interface(**source_data[name])
-            for name, data_interface in self.data_interface_classes.items()
-            if name in source_data
-        }
+
+        assert_msg = "Received both data_interfaces and source_data passed as arguments. Specify only one of them"
+        assert source_data is None or data_interfaces is None, assert_msg
+
+        if source_data is not None:
+            self._validate_source_data(source_data=source_data, verbose=self.verbose)
+
+            self.data_interface_objects = {
+                name: data_interface(**source_data[name])
+                for name, data_interface in self.data_interface_classes.items()
+                if name in source_data
+            }
+
+        elif data_interfaces is not None:
+            if isinstance(data_interfaces, list):
+                # Create unique names for each interface
+                counter = {interface: 0 for interface in data_interfaces}
+                total_counts = Counter(data_interfaces)
+                interface_names = dict()
+                for interface in data_interfaces:
+                    counter[interface] += 1
+                    unique_signature = f"{counter[interface]:03}" if total_counts[interface] > 1 else ""
+                    interface_name = f"{interface.__class__.__name__}{unique_signature}"
+                    interface_names[interface] = interface_name
+
+                self.data_interface_objects = {name: interface for interface, name in interface_names.items()}
+            else:
+                self.data_interface_objects = data_interfaces
+
+            self.data_interface_classes = {name: interface.__class__ for interface, name in interface_names.items()}
```

Comment:
The keys in this dict need to match the keys of the `data_interface_objects` dict